### ISSUE TITLE
`STAGE:[resource name]` should return that resources AggregateResource structure, not simply that resources amount

### ISSUE THREAD

**Issue by space-is-hard** [KSP-KOS/KOS#1239]
I realize that this would be a *very* breaking change, as I'm sure that many scripts out there rely on things like `STAGE:LIQUIDFUEL` returning the amount for staging checks, however the current implementation is, in my opinion, counterintuitive. It prevents the user from being able to use all of the suffixes of the AggregateResource structure.

**Comment by hvacengi** [comment]
What about offering a suffix of `STAGE:RESOURCESLEX` which contains a dictionary of resources?  That would allow you to query `STAGE:RESOURCESLEX["LiquidFuel"]`.  We could also alias it to something shorter like `RESLEX`.  Ideally, I'd rather replace the list with the dictionary, but a new suffix gives easy backwards compatibility.

Actually, we should take a long look at any other list suffixes in the code base.  I bet there are a bunch that we could offer as lexicon objects to help users who currently have to iterate through each checking the `NAME` suffix.

**Comment by tj-murphy** [comment]
Yeah, I've seen this issue too. STAGE:LIQUIDFUEL:CAPACITY fails because kOS evaluates "STAGE:LIQUIDFUEL" into its value. Then it tries to take this number and check the capacity of a number which makes no sense. I think the KOS core needs to have two different forms of "STAGE:LIQUIDFUEL", one of which is for one without a further suffix and one for the "naked" use of "STAGE:LIQUIDFUEL". The form that is being used would be evaluated on the fly.
